plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaFilterManager.java

```diff
@@ -129,11 +129,15 @@ public KafkaFilteringResult getKafkaFilterResult(
             if (offsetTimestampRanged.isPresent()) {
                 try (KafkaConsumer<byte[], byte[]> kafkaConsumer = consumerFactory.create(session)) {
                     Optional<Range> finalOffsetTimestampRanged = offsetTimestampRanged;
-                    partitionBeginOffsets = overridePartitionBeginOffsets(partitionBeginOffsets,
-                            partition -> findOffsetsForTimestampGreaterOrEqual(kafkaConsumer, partition, finalOffsetTimestampRanged.get().getBegin()));
+                    if (offsetTimestampRanged.get().getBegin() >= 0) {
```

Comment:
Can we add some tests to check if it works as expected ?

Reply:
OK, I added some test cases in TestKafkaIntegrationPushDown.